libs/coin-modules/coin-evm/src/__tests__/unit/broadcast.unit.test.ts

```diff
@@ -64,16 +85,117 @@ describe("EVM Family", () => {
+        });
+
+        afterEach(() => {
+          jest.clearAllMocks();
+        });
+
+        afterAll(() => {
+          mockBroadcastTransactions();
+        });
+
+        const coinTransaction: EvmTransaction = {
+          amount: new BigNumber(100),
+          useAllAmount: false,
+          subAccountId: "id",
+          recipient: "0x51DF0aF74a0DBae16cB845B46dAF2a35cB1D4168", // michel.eth
+          feesStrategy: "custom",
+          family: "evm",
+          mode: "send",
+          nonce: 0,
+          gasLimit: new BigNumber(21000),
+          chainId: 1,
+          maxFeePerGas: new BigNumber(100),
+          maxPriorityFeePerGas: new BigNumber(100),
+          type: 2,
+        };
+        const broadcastArgs = {
+          account,
+          signedOperation: {
+            operation: buildOptimisticOperation(account, coinTransaction),
+            signature: "0xS1gn4tUR3",
+          },
+        };
+
+        it("Ledger node MEV ON/OFF", async () => {
+          mockGetConfig.mockImplementation((): any => ({
+            info: {
+              node: {
+                type: "ledger",
+              },
+            },
+          }));
+
+          mockAxios.request.mockResolvedValue({ data: mockedBroadcastResponse });
+          const axiosSpy = jest.spyOn(mockAxios, "request");
+
+          // MEV OFF
+          await broadcast({
+            ...broadcastArgs,
+            broadcastConfig: { mevProtected: false },
+          });
+
+          expect(axiosSpy).toHaveBeenCalled();
+
+          const requestConfigOff = axiosSpy.mock.calls[0][0] as { params: any };
+          const urlParamsOff = new URLSearchParams(requestConfigOff.params).toString();
+
+          // MEV ON
+          await broadcast({
+            ...broadcastArgs,
+            broadcastConfig: { mevProtected: true },
+          });
+
+          expect(axiosSpy).toHaveBeenCalledTimes(2);
+
+          const requestConfigOn = axiosSpy.mock.calls[1][0] as { params: any };
+          const urlParamsOn = new URLSearchParams(requestConfigOn.params).toString();
+
+          expect(urlParamsOff).toContain("mevProtected=false");
+          expect(urlParamsOn).toContain("mevProtected=true");
+        });
+
+        it("External node MEV ON/OFF", async () => {
+          mockGetConfig.mockImplementation((): any => ({
+            info: {
+              node: {
+                type: "external",
+                uri: "https://my-rpc.com",
+              },
+            },
+          }));
+
+          const providerSpy = jest.spyOn(mockEthers.providers, "StaticJsonRpcProvider");
+
+          // MEV OFF
+          await broadcast({
+            ...broadcastArgs,
+            broadcastConfig: { mevProtected: false },
+          });
+
+          // MEV ON
+          await broadcast({
+            ...broadcastArgs,
+            broadcastConfig: { mevProtected: true },
+          });
+
+          expect(providerSpy).toHaveBeenCalledTimes(1);
```

Comment:
Almost all good, you're just missing the important part of this one which is testing that providerSpy is called with `"https://my-rpc.com"` both times. 
If we were to implement MEV protection for light integrations, we would switch node here and broadcast with MEV protected one. The point of the test is to demonstrate that it's not the case for now, we're maintaining everything as is.

Reply:
![image](https://github.com/user-attachments/assets/9f3e4358-5719-41aa-9d56-780a2fc0964b)

Reply:
You're right !  `expect(providerSpy).toHaveBeenCalledWith(1, "https://my-rpc.com");` should be good enough then with the current behavior, as if we were to change that to use another RPC, we'd have to call it 2 times minimum so it would be covered 👌

Reply:
That works ! Thanks.